### PR TITLE
Binding

### PR DESCRIPTION
**Pull request by mdinger**
In [this reddit thread](http://www.reddit.com/r/rust/comments/2zt059/very_newb_match_and_binding/), there was confusion over the purpose of `match` binding. I expanded the example to the thread but no comments were made. This is a follow-up to be more clear about it's purpose.

It also uses `u32` to disallow bizarre results like "I'm a -4 year old person".

### CODE REVIEW

**Comment by mdinger** [comment]
Repushed to get travis to retry

File: examples/flow_control/match/binding/input.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
-Rust supports binding variables via a `match`:
+Indirectly accessing a variable makes it impossible to branch and use that
+variable without re-binding. `match` provides the `@` sygil for exactly
```
Comment (steveklabnik) [inline examples/flow_control/match/binding/input.md]: sigil

**Comment by steveklabnik** [comment]
Looks good, other than that one nit

File: examples/flow_control/match/binding/input.md (hunk 1 above)
Comment (mdinger) [inline examples/flow_control/match/binding/input.md]: Thanks. I remember typing that but thought I fixed it.

**Comment by mdinger** [comment]
This is updated too.

**Pull request merged**